### DISCUSSION TITLE
How to set a boolean after a command is used?

### DISCUSSION THREAD

**Discussion by BGPCS** [Discord4J/Discord4J#1012] [Q&A] [answered]
Hi, I'm a bit of an idiot and trying to learn how to make an RPG discord bot. 

I'm trying to set the bot so that when someone does !battle, it sends a message (that part's working), and then sets the boolean InBattle = true. Issue is, I'll try to add InBattle, and no matter where I put it, it seems to error out. Any idea where I'm intended to put it? 

I

        client.getEventDispatcher().on(MessageCreateEvent.class)
                .map(MessageCreateEvent::getMessage)
                .filter(message -> message.getAuthor().map(user -> !user.isBot()).orElse(false))
                .filter(message -> message.getContent().equalsIgnoreCase("!battle"))
                .flatMap(Message::getChannel)
                .flatMap(channel -> channel.createMessage("<:walt:751994064399892512> appears!\nHis cancer ridden body has " + testhp + " hp!\nWhat will you do?"))
                .subscribe();

**Comment by NovaFox161** [comment, marked as answer]
I'd recommend joining the support server linked in the readme and asking in the dedicated channel for d4j-help or programming-help.

https://discord.gg/d4j
